app/src/main/java/org/simple/clinic/scanid/ScanSimpleIdUpdate.kt

```diff
@@ -39,12 +44,19 @@ class ScanSimpleIdUpdate @Inject constructor(
       is ScanSimpleIdScreenQrCodeScanned -> simpleIdQrScanned(model, event)
       is PatientSearchByIdentifierCompleted -> patientSearchByIdentifierCompleted(model, event)
       is ScannedQRCodeJsonParsed -> scannedQRCodeParsed(model, event)
-      InvalidQrCode -> next(model.notSearching().invalidQrCode())
+      InvalidQrCode -> invalidQrCode(model)
       is OnlinePatientLookupWithIdentifierCompleted -> onlinePatientLookupWithIdentifierCompleted(model, event)
       is CompleteMedicalRecordsSaved -> patientsFoundByOnlineLookup(event.completeMedicalRecords)
     }
   }
 
+  private fun invalidQrCode(model: ScanSimpleIdModel): Next<ScanSimpleIdModel, ScanSimpleIdEffect> {
+    return when (model.openedFrom) {
+      is EditPatientScreen -> dispatch(ShowScannedQrCodeError(ScanErrorState.InvalidQrCode))
```

Comment:
Yep, just like this. Let's do this for other places as well.

I think we don't specifically need to have subclasses of this to like `ToAddNHID` or `ToAddBpPassport`. Since we will be returning the identified directly, and we already have code to identifiy if the scanned QR is BP Passport or NHID (the try/catch block)